draft-ietf-ipsecme-rfc8229bis.xml

```diff
@@ -0,0 +1,1373 @@
+          session has already been established and the Initiator's IP address
+          has been verified.  Moreover, a TCP/IP stack  creates a state once
+          a TCP SYN packet is received (unless SYN Cookies described in <xref target="RFC4987" />
+          are employed), which contadicts the statelessness of IKEv2 Cookies.
+          In particular, with TCP an attacker is able to mount a SYN flooding DoS attack 
+          which IKEv2 Responder cannot prevent using stateless IKEv2 Cookies.
+          So, when using TCP encapsulation it makes little sense to send Cookie requests without 
+          Puzzles unless the Responder is concerned with a possibility of TCP
+          Sequence Number attacks (see <xref target="RFC6528" /> for details). Puzzles, on
+		  the other hand, still remain useful (and their use requires using Cookies).
+          </t>
+
+          <t>The following considerations are applicable for using Cookie and
+          Puzzle mechanisms in case of TCP encapsulation:
+          <list style="symbols" >
+            <t>the exchange Responder SHOULD NOT request a Cookie, with the
+			exception of Puzzles or in rare cases like preventing TCP Sequence
+			Number attacks.
+            </t>
+            <t>if the Responder chooses to send Cookie request (possibly along
+            with Puzzle request), then the TCP connection that the IKE_SA_INIT
+            request message was received over SHOULD be closed, so that the
+            Responder remains stateless at least until the Cookie (or Puzzle
+            Solution) is returned. Note that if this TCP connection is closed,
+			the Responder MUST NOT include the Initiator's TCP port into the Cookie
+			calculation (*), since the Cookie will be returned over a new
+			TCP connection with a different port.
+            </t>
+            <t>the exchange Initiator acts as described in Section 2.6 of
+            <xref target="RFC7296" /> and Section 7 of <xref target="RFC8019" />, 
+            i.e. using TCP encapsulation doesn't change the Initiator's behavior.
+            </t>
+          </list>
+
+          (*) Examples of Cookie calculation methods are given in Section 2.6
+          of <xref target="RFC7296" /> and in Section 7.1.1.3 of <xref target="RFC8019" /> 
+          and they don't include transport protocol ports.  However these examples are given
+          for illustrative purposes, since Cookie generation algorithm is a
+          local matter and some implementations might include port numbers,
+          that won't work with TCP encapsulation. Note also, that these 
+          examples include Initiator's IP address in Cookie calculation.
+          In general this address may cahnge between two initial requests (w/o and with Cookie).
+          This may happen due to NATs in between, since in case of TCP protocol NATs have more freedom 
+          in changing source IP address for new TCP connections. In this case cookie verification would fail.
```

Comment:
```suggestion
          This may happen due to NATs, since NATs have more freedom to change change source IP addresses for new
          TCP connections than for UDP. In such cases, cookie verification might fail.
```